### ISSUE TITLE
New sugestions

### ISSUE THREAD

**Issue by pmiguelr** [falsovsky/chaturbate#1]
Hi..

I have some sugestions that i would have on that script. I don't know how implemented it.
However i created my apache site with that.. i run bash file when i need.

Example:

movie_preview.sh file to create a 100 frames thumb file from a .mp4 file
#!/bin/bash

if [ -z "$1" ]; then
    echo "usage: ./movie_preview.sh VIDEO [HEIGHT=120] [COLS=100] [ROWS=1] [OUTPUT]"
    exit
fi

MOVIE=$1
HEIGHT=$2
COLS=$3
ROWS=$4
OUT_FILENAME=$5

# get video name without the path and extension
MOVIE_NAME=`basename $MOVIE`
OUT_DIR=`pwd`

if [ -z "$HEIGHT" ]; then
    HEIGHT=120
fi
if [ -z "$COLS" ]; then
    COLS=100
fi
if [ -z "$ROWS" ]; then
    ROWS=1
fi
if [ -z "$OUT_FILENAME" ]; then
    OUT_FILENAME=`echo ${MOVIE_NAME%.*}_preview.jpg`
fi

OUT_FILEPATH=`echo $OUT_DIR/$OUT_FILENAME`

TOTAL_IMAGES=`echo "$COLS*$ROWS" | bc`

# get total number of frames in the video
# ffprobe is fast but not 100% reliable. It might not detect number of frames correctly!
NB_FRAMES=`ffprobe -show_streams "$MOVIE" 2> /dev/null | grep nb_frames | head -n1 | sed 's/.*=//'`

if [ "$NB_FRAMES" = "N/A" ]; then
    # as a fallback we'll use ffmpeg. This command basically copies this
    # video to /dev/null and it counts frames in the process.
    # It's slower (few seconds usually) than ffprobe but works everytime.
    NB_FRAMES=`ffmpeg -nostats -i "$MOVIE" -vcodec copy -f rawvideo -y /dev/null 2>&1 | grep frame | awk '{split($0,a,"fps")}END{print a[1]}' | sed 's/.*= *//'`
fi

# calculate offset between two screenshots, drop the floating point part
NTH_FRAME=`echo "$NB_FRAMES/$TOTAL_IMAGES" | bc`
echo "capture every ${NTH_FRAME}th frame out of $NB_FRAMES frames"

# make sure output dir exists
mkdir -p $OUT_DIR

FFMPEG_CMD="ffmpeg -loglevel panic -i \"$MOVIE\" -y -frames 1 -q:v 1 -vf \"select=not(mod(n\,$NTH_FRAME)),scale=-1:${HEIGHT},tile=${COLS}x${ROWS}\" \"$OUT_FILEPATH\""

eval $FFMPEG_CMD
echo $OUT_FILEPATH

----
After that html file with css to see animation of thumb (100 frames) from that movie.

<a href='#' target='_blank' alt='' title='' class='video-preview' data-frames='100' data-source='preview.jpg'></a>
<style>
body {
    text-align: center;
    padding-top: 20px;
}

.video-preview {
    display: inline-block;
    position: relative;
    background: #ddd;
    overflow: hidden;
    /* This is temporary width and height, these'll be overriden when the source img is loaded.*/
    /* If you already know size of a preview frame you can hardcode it here. */
    width: 160px;
    height: 120px;
    border-radius: 3px;
    box-shadow: 0 0 6px #bbb;
}
</style>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
<script>
(function($) {
    $.fn.videoPreview = function(options) {
        return this.each(function() {
            var elm = $(this);
            var frames = parseFloat(elm.data('frames'));

            var img = $('<img/>', { 'src': elm.data('source') }).hide().css({
                'position': 'absolute', 'cursor': 'pointer'
            }).appendTo(elm);
            var slider = $('<div/>').hide().css({
                'width': '2px', 'height': '100%', 'background': '#ddd', 'position': 'absolute',
                'z-index': '1', 'top': '0', 'opacity': 0.6, 'cursor': 'pointer'
            }).appendTo(elm);

            var width;

            function defaultPos() {
                img.css('left', -width * frames / 4);
            }

            img.load(function() { // we need to know video's full width
                $(this).show();
                width = this.width / frames;
                elm.css('width', width);
                defaultPos();
            });
            elm.mousemove(function(e) {
                var left = e.clientX - elm.position().left; // position inside the wrapper
                slider.show().css('left', left - 1); // -1 because it's 2px width
                img.css('left', -Math.floor((left / width) * frames) * width);
            }).mouseout(function(e) {
                slider.hide();
                defaultPos();
            });

        });
    };
})(jQuery);

$('.video-preview').videoPreview();
</script>

**Comment by falsovsky** [comment]
You can use this tool for this.

http://moviethumbnail.sourceforge.net/

**Comment by falsovsky** [comment]
Well it seems it isnt no longer maintained. I couldnt even build it.

I'm going to take a look at your script :-)

**Comment by falsovsky** [comment]
I almost have this ready :-)
